Travis failed with:
```
pepper_cli = <function _run_pepper_cli at 0x7f0f78abbc08>
session_minion_id = 'pytest-session-salt-minion-0'
    def test_local_poll(pepper_cli, session_minion_id):
        '''Test the returns poller for localclient'''
        ret = pepper_cli('--run-uri', '--fail-if-incomplete', '*', 'test.sleep', '30')
>       assert ret[0][session_minion_id] is True
E       TypeError: string indices must be integers, not str
tests/integration/test_poller.py:10: TypeError
```
But I see no relation with this particular patch. session_minion_id is str... I have no idea about it. Should I fix something?

--- a/pepper/cli.py
+++ b/pepper/cli.py
@@ -326,30 +326,30 @@ class PepperCli(object):
 
         return optgroup
 
-
     def add_retcodeopts(self):
         '''
         ret code validation options
         '''
-        optgroup = optparse.OptionGroup(self.parser, "retcode Field Validation Options",
-                textwrap.dedent("""\
-                Validate return.HOST.retcode fields
-                """))
+        optgroup = optparse.OptionGroup(
+            self.parser, "retcode Field Validation Options", "Validate return.HOST.retcode fields")
 
-        optgroup.add_option('--fail-any', dest='fail_any', action='store_true',
-                help="Fail if any of retcode field is non zero.")
+        optgroup.add_option(
+            '--fail-any', dest='fail_any', action='store_true',
+            help="Fail if any of retcode field is non zero.")
 
-        optgroup.add_option('--fail-any-none', dest='fail_any_none', action='store_true',
-                help="Fail if any of retcode field is non zero or there is no retcode at all.")
+        optgroup.add_option(
+            '--fail-any-none', dest='fail_any_none', action='store_true',
+            help="Fail if any of retcode field is non zero or there is no retcode at all.")
 
-        optgroup.add_option('--fail-all', dest='fail_all', action='store_true',
-                help="Fail if all retcode fields are non zero.")
+        optgroup.add_option(
+            '--fail-all', dest='fail_all', action='store_true',
+            help="Fail if all retcode fields are non zero.")
 
-        optgroup.add_option('--fail-all-none', dest='fail_all_none', action='store_true',
-                help="Fail if all retcode fields are non zero or there is no retcode at all.")
+        optgroup.add_option(
+            '--fail-all-none', dest='fail_all_none', action='store_true',
+            help="Fail if all retcode fields are non zero or there is no retcode at all.")
 
         return optgroup
-
 
     def get_login_details(self):
         '''

--- a/pepper/retcode.py
+++ b/pepper/retcode.py
@@ -67,7 +67,7 @@ class PepperRetcode(object):
                                 for name in minion if isinstance(minion[name], dict) and
                                 minion[name].get('retcode') is not None)
                 if not retcodes:
-                    return -1 # there are no retcodes
+                    return -1  # there are no retcodes
                 return next((r for r in retcodes if r != 0), 0)
         return -1
 
@@ -110,7 +110,7 @@ class PepperRetcode(object):
                                 for name in minion if isinstance(minion[name], dict) and
                                 minion[name].get('retcode') is not None)
                 if not retcodes:
-                    return -1 # there are no retcodes
+                    return -1  # there are no retcodes
                 if all(r for r in retcodes if r != 0):
                     return next((r for r in retcodes if r != 0), 0)
         return -1

--- a/pepper/script.py
+++ b/pepper/script.py
@@ -90,7 +90,8 @@ class Pepper(object):
                     else:
                         print(result)
                 if exit_code is not None:
-                    exit_code = PepperRetcode().validate(result)
+                    if exit_code == 0:
+                        return PepperRetcode().validate(result)
                     return exit_code
         except PepperException as exc:
             print('Pepper error: {0}'.format(exc), file=sys.stderr)

--- a/pepper/script.py
+++ b/pepper/script.py
@@ -89,9 +89,8 @@ class Pepper(object):
                             print(result, file=ofile)
                     else:
                         print(result)
-                if exit_code is None or exit_code == 0:
-                    exit_code = PepperRetcode().validate(result)
                 if exit_code is not None:
+                    exit_code = PepperRetcode().validate(result)
                     return exit_code
         except PepperException as exc:
             print('Pepper error: {0}'.format(exc), file=sys.stderr)

